source/s3_client.c

```diff
@@ -365,6 +359,14 @@ struct aws_s3_client *aws_s3_client_new(
         }
     }
 
+    if (client_config->proxy_options) {
+        client->proxy_config = aws_http_proxy_config_new_from_proxy_options_with_tls_info(
```

Comment:
trivial: you can use `client_config->tls_mode == AWS_MR_TLS_ENABLED` instead of `client->tls_connection_options != NULL`